src/components/DataTypes/array.tsx

```diff
@@ -190,25 +143,114 @@ export const getFilter = (schema: JSONSchema, value: string) => {
 		return {
 			minItems: 1,
 			items: {
-				properties: Object.keys(schema.items.properties).reduce(
-					(props: NonNullable<JSONSchema['properties']>, propKey: string) => {
-						props[propKey] = {
-							// @ts-expect-error
-							regexp: {
-								pattern: regexEscape(value),
-								flags: 'i',
+				anyOf: [
+					...Object.entries(schema.items.properties).reduce(
+						(props: any[], [propKey, propValue]: [string, JSONSchema]) => {
+							const subSchema = isFullTextSearch
+								? { ...propValue, oneOf: undefined, enum: undefined }
+								: propValue;
+							const model = getDataModel(subSchema);
+							const operators = model?.operators(schema);
+							const newOperator = convertOperator(operator, operators);
+							const filter = model?.createFilter(
+								key,
+								propKey,
+								newOperator,
+								value,
+								subSchema,
+								true,
+							);
+							props.push(filter);
+							return props;
+						},
+						[],
+					),
+				],
+			},
+		};
+	} else if (
+		!!schema?.items &&
+		typeof schema.items !== 'boolean' &&
+		'type' in schema.items
+	) {
+		const filter = getModelFilter(
+			schema.items,
+			key,
+			field,
+			operator,
+			value,
+			isFullTextSearch,
+			convertOperator,
+		);
+		if (filter.properties?.[field]) {
+			return { contains: filter.properties?.[field] as JSONSchema };
+		}
+		return {
+			anyOf: [
+				{
+					properties: {
+						[field]: {
+							type: 'array',
+							description: value,
+							contains: {
+								not: {
+									const: value,
+								},
 							},
-						};
-						return props;
+						},
 					},
-					{},
-				),
-			},
+				},
+				{
+					properties: {
+						[field]: {
+							type: 'null',
+						},
+					},
+				},
+			],
 		};
 	}
-	return {
-		contains: {
-			const: Number.isNaN(parseInt(value, 10)) ? value : Number(value),
-		},
-	};
+	const filter = getModelFilter(
+		schema,
+		key,
+		field,
+		operator,
+		value,
+		isFullTextSearch,
+		convertOperator,
+	);
+	return { contains: filter };
+};
+
+const convertOperator = (operator: Operator, operators: any) => {
+	if (operator.slug in operators) {
+		return operator;
+	}
+	if (operator.label.includes('not')) {
+		return { slug: 'is_not', label: 'is not' };
+	}
+	return { slug: 'is', label: 'is' };
+};
+
+export const editSchema = (
+	schema: JSONSchema,
+	operator: Operator | undefined,
+): JSONSchema => {
+	const refScheme = getPropertyRefScheme(schema);
+	const schemaItems = schema.items as JSONSchema | undefined;
+	const convertedRefScheme = convertRefSchemeToSchemaPath(refScheme);
+	if (!!schemaItems && !!convertedRefScheme) {
+		const property = get(
+			schemaItems.properties,
+			convertedRefScheme,
+		) as JSONSchema['properties'];
+		return property as JSONSchema;
+	}
+	const subSchema = schemaItems || schema;
+	const model = getDataModel(subSchema);
+	return (
+		model && 'editSchema' in model
+			? model.editSchema(subSchema, operator)
+			: subSchema
+	) as JSONSchema;
```

Comment:
Looks like we only need to cast the `model.editSchema` part, not the entire ternary